doc/devel/contributing.rst

```diff
@@ -595,3 +595,52 @@ example code you can load it into a file handle with::
 
     import matplotlib.cbook as cbook
     fh = cbook.get_sample_data('mydata.dat')
+
+
+Baseline image generation
+-------------------------
+The first time this test is run, there will be no baseline image to compare against, so the test will fail. The baseline
+images would be generated either by installing ``matplotlib-baseline-images`` package or by running the baseline image
+generation command from the commandline. These methods are described below. Later on, developer can run ::
```

Comment:
Should we be suggesting folks generate the images themselves?  That will introduce all sorts of changes due to bad font installs.  I think tests should be done against canonical, even locally.

Reply:
The point of this work is so that we _don't_ have to compare against the canonical version of the images which eliminates the issues we have with freetype / latex drifting under us.  So long as you don't change your local images commit-to-commit they should also not change commit-to-commit on a different machine with a different freetype / latex install.

The plan is to with a release publish a set of "blessed" images with a fixed freetype.

Reply:
OK, I thought the point was not to have all the images in the repo.  

So the idea is that I generate the images off of master, and then make a new branch, make my changes and then run the tests and compare my changes to the ones I made on master.  

A possible difficulty here are all the fixed-size tests we have that are *not* image tests.  There are many tests we have that assert the position of an element is at 0.9765, but it may not be if we don't have the same font library.  

In either case, the above description is not clear on the process.  I'd say something like

```
In order to run local image tests, the developer should checkout the ``master`` branch and run the tests once to get a set of baseline images that are consistent with the code in ``master``.  Then subsequent tests from their branches will test against these baseline images.  

Alternately, the developer could download the ``matplotlib-baseline-images`` package and compare against these images.  However, the developer will have to be certain to have  development environment that has the same free type and other dependencies as the testing environment on the the machine that made ``matplotlib-baseline-images``.  

```

Reply:
The idea is to eventually remove all of the images but we have options about where devs get the baselines from.